### DISCUSSION TITLE
Building IfcOpenshell in Visual Studio 2022  with vs2019-x86 generator settings

### DISCUSSION THREAD

**Discussion by SP-AWN** [IfcOpenShell/IfcOpenShell#5270] [Q&A]
Hi,
I have Visual Studio Community 2022 installed but I need library to be compiled with VS2019. I have msvc 142 compilation tools installed, but I don't have Tools Command Prompt for VS 2019 available, so all I can is run build-deps.cmd under VS 2022. Any suggestions how to get library in required version?
Thank in advance for help
Regards
Adam

**Comment by RickBrice** [comment]
Sorry I don’t have an answer to your specific question, but visual studio is pretty good with binary compatibility. Have you tried building with vs2022 and then trying to call it with your vs2019 code? It might work.

**Reply by SP-AWN** [reply]
Hi,
I will try it. 
Thanks for advice.

**Comment by aothms** [comment]
You might also want to consider installing through chocolatey https://community.chocolatey.org/packages/visualstudio2019buildtools

It wouldn't give you the IDE, but for that you can use the 2022 community edition then.
